### DISCUSSION TITLE
How to change the ui of wazuh dashboard from it's code? Please help.

### DISCUSSION THREAD

**Discussion by Prathamesh314** [wazuh/wazuh-kibana-app#7188] [Q&A] [answered]
I want to change ui of wazuh dashboard and for that I followed these steps:

1. clone the wazuh-dashboard repository
2. After cloning, I cloned wazuh-dashboard-plugins repository inside plugins folder of wazuh-dahboard
3. I made changes to some texts in wazuh modules and updated all the images
4. I packaged the whole repository which is stored in dev-tools/build-packages/output/deb folder
5. I used offline component installation to setup wazuh and there I used my generated dashboard package and for the rest I used in-built packages, i.e. for indexer and manager.
6. When I started dashboard on browser, only images changed not the text. Please help me where and how to change the ui

**Comment by Tostti** [comment]
Hi @Prathamesh314 
I understand that you were able to successfully generate the package, but the changes are not working. 

Most of the possible changes are inside the `wazuh-dashboard-plugins` repo. In this repo, you can navigate to `/docker/osd-dev` and use that to bring up a development environment on which you will be able to see all the changes you do without generating packages. 

To do this you will need docker, and Node.js in the version that the repository states in `.nvmrc`. Once you have that, you can bring up the environment with `./dev.sh /path/to/wazuh-dashboard-plugins/plugins up`. You can also add `server 4.9.2` at the end to bring up a manager in the specified version. Once all the containers are up, attach a shell to the osd container and run `yarn start --no-base-path`. After that, you will be able to login to `localhost:5601` with the credentials `admin:admin`. 

If the changes you need to do are in the `wazuh-dashboard` repository, you can similarly navigate to `docker` and run `./dev.sh up` to bring up the developement environment for that repository. 

I hope you find this helpful, let me know if you need further assistance.

Regards,

**Reply by Prathamesh314** [reply]
I check the docker networks. They are running.
Issues I am facing is:

 When I use this path: /home/vgadmin/wazuh/wazuh-dashboard/plugins/wazuh-dashboard-plugins/plugins and execute "yarn start" inside container, then I am getting an error: 
        log   [05:12:54.964] [error][data][opensearch] [ConnectionError]: unable to verify the first certificate
server   log   [05:12:55.264] [error][data][opensearch] [ConnectionError]: unable to verify the first certificate
 log   [05:12:57.456] [error][data][opensearch] [ConnectionError]: unable to verify the first certificate
server   log   [05:12:57.648] [error][data][opensearch] [ConnectionError]: unable to verify the first certificate
 log   [05:12:59.893] [error][data][opensearch] [ConnectionError]: unable to verify the first certificate
server   log   [05:13:00.210] [error][data][opensearch] [ConnectionError]: unable to verify the first certificate


2. When I use this path: /home/vgadmin/wazuh/wazuh-dashboard/plugins/wazuh-dashboard-plugins and replicate same procedure, I am getting this error:
np bld  log  [05:27:12.543] [success][@osd/optimizer] 51 bundles compiled successfully after 334.9 sec, watching for changes
server   error  [05:33:59.392] [error][client][connection] Error: 40DC7B4525700000:error:0A000416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1586:SSL alert number 46

server   error  [05:33:59.751] [error][client][connection] Error: 40DC7B4525700000:error:0A000416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1586:SSL alert number 46

server   error  [05:34:02.391] [error][client][connection] Error: 40DC7B4525700000:error:0A000416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1586:SSL alert number 46

server   error  [05:34:04.552] [error][client][connection] Error: 40DC7B4525700000:error:0A000416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1586:SSL alert number 46

server   error  [05:34:04.553] [error][client][connection] Error: 40DC7B4525700000:error:0A000416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1586:SSL alert number 46

server   error  [05:34:07.942] [error][client][connection] Error: 40DC7B4525700000:error:0A000416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1586:SSL alert number 46

server   error  [05:34:07.946] [error][client][connection] Error: 40DC7B4525700000:error:0A000416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1586:SSL alert number 46

server   error  [05:34:08.436] [error][client][connection] Error: 40DC7B4525700000:error:0A000416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1586:SSL alert number 46

server   error  [05:34:08.438] [error][client][connection] Error: 40DC7B4525700000:error:0A000416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1586:SSL alert number 46

server   error  [05:34:09.111] [error][client][connection] Error: 40DC7B4525700000:error:0A000416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1586:SSL alert number 46

and on the browser i am seeing:  "Application not found"

3. When I am using this path: /home/vgadmin/wazuh/wazuh-dashboard/plugins/
I am getting this error: [agentkeepalive:deprecated] options.freeSocketKeepAliveTimeout is deprecated, please use options.freeSocketTimeout instead
server   log   [05:50:00.645] [error][cron-scheduler][plugins][wazuh] TypeError [ERR_INVALID_URL]: Invalid URL
server   log   [05:50:00.649] [error][cron-scheduler][plugins][wazuh] TypeError [ERR_INVALID_URL]: Invalid URL
server   log   [05:50:00.653] [error][cron-scheduler][plugins][wazuh] Error: getaddrinfo ENOTFOUND imposter
server   log   [05:50:00.654] [error][cron-scheduler][plugins][wazuh] Error: getaddrinfo ENOTFOUND wazuh.manager
server   log   [05:50:00.655] [error][cron-scheduler][plugins][wazuh] Error: getaddrinfo ENOTFOUND wazuh.manager
server   log   [05:50:00.656] [error][cron-scheduler][plugins][wazuh] Error: getaddrinfo ENOTFOUND imposter
server   log   [05:55:00.877] [error][cron-scheduler][plugins][wazuh] TypeError [ERR_INVALID_URL]: Invalid URL
server   log   [05:55:00.891] [error][cron-scheduler][plugins][wazuh] TypeError [ERR_INVALID_URL]: Invalid URL
server   log   [05:55:00.907] [error][cron-scheduler][plugins][wazuh] Error: getaddrinfo ENOTFOUND imposter
server   log   [05:55:00.911] [error][cron-scheduler][plugins][wazuh] Error: getaddrinfo ENOTFOUND wazuh.manager
server   log   [05:55:00.914] [error][cron-scheduler][plugins][wazuh] Error: getaddrinfo ENOTFOUND imposter
server   log   [05:55:00.915] [error][cron-scheduler][plugins][wazuh] Error: getaddrinfo ENOTFOUND wazuh.manager

and on the browser i am seeing exactly same as previous thread's image. everything disabled.

**Reply by Tostti** [reply]
I understand. The `application not found` is because the plugins are not being read, so the directory should be `<path>/wazuh-dashboard-plugins/plugins`.

I will try to replicate the issue to find a solution. Could you tell me which branch or tag of  `wazuh-dashboard-plugins` you use for this? Note that versions above 4.9.2 are not yet released, and could have several errors or even not work.

**Reply by Prathamesh314** [reply]
I am using tag v4.9.2

**Reply by Tostti** [reply]
That tag should be ok. 

Did you create the required docker networks? You can see it in the [pre-requisites](https://github.com/wazuh/wazuh-dashboard-plugins/tree/v4.9.2/docker#pre-requisites) of the docker environments. (The 3rd step on that documentation is not required)

**Reply by Prathamesh314** [reply]
I have followed every step given in that documenation, I am seeing this page on browser:

when I am using this path: "/home/vgadmin/wazuh/wazuh-dashboard/plugins/" with dev.sh cmd
<img width="1470" alt="Screenshot 2024-12-06 at 6 44 35 PM" src="https://github.com/user-attachments/assets/b7e30886-313b-4bf2-bd9a-5c80cd44ac97">


when I am using this path: "/home/vgadmin/wazuh/wazuh-dashboard/plugins/wazuh-dashboard-plugins"

<img width="1470" alt="Screenshot 2024-12-06 at 1 04 45 PM" src="https://github.com/user-attachments/assets/7722d1cc-a84c-4d60-8ee7-ccc86b2fdb05">

**Reply by Prathamesh314** [reply]
You can help me in one way.

You try to run wazuh in development environment using docker from scratch and provide me the commands you executed.

**Reply by Tostti** [reply, marked as answer]
In the first image that you show, the environment is working as expected. That API screen is part of the Wazuh plugin, meaning that is being correctly read.

Regarding the API errors, I found the cause of the error, that was due to a wrong IP in the imposter configuration file. To solve it, open `wazuh-dashboard-plugins/docker/imposter/wazuh-config.yml` and replace 

```
specFile: https://raw.githubusercontent.com/wazuh/wazuh/4.9.2/api/api/spec/spec.yaml
```
with 
```
specFile: https://raw.githubusercontent.com/wazuh/wazuh/v4.9.2/api/api/spec/spec.yaml
```
(add a `v` before the Wazuh version). After that, you can start the imposter container again and it should work. 

There's also the option to set a real manager (imposter is a mock of the API). To do that, you need to add `server 4.9.2` at the end of the `dev.sh` call to bring up the environments. For example `./dev.sh $WZ_HOME up server 4.9.2`. 

I hope this solves your issue

**Reply by Prathamesh314** [reply]
thanks @Tostti . It worked.

**Comment by rhs199408** [comment]
Hi, @Prathamesh314
I'm also trying to change the ui of wazuh dashboard, both of text and images, just like you.
But I have no idea.
Please explain how you did it step by step.

**Comment by Mubashirmuchi** [comment]
Successfully set up and configured the customized Wazuh Dashboard environment, resolving all dependency and bootstrap issues. The dashboard runs correctly as a standalone instance. The **wazuh-dashboard-plugins** repository has been cloned; however, integration is currently failing during installation. The base dashboard remains fully functional without the plugins, and troubleshooting is ongoing to identify the root cause.
